proto/regen/ecocredit/v1alpha2/types.proto

```diff
@@ -225,18 +225,54 @@ message AskDenom {
   uint32 exponent = 3;
 }
 
+// BasketCredit represents the information for a credit batch inside a basket.
+message BasketCredit {
+  
+  // batch_denom is the unique ID of the credit batch.
+  string batch_denom = 1;
+
+  // tradable_amount is the number of credits in this transfer that can be
+  // traded by the recipient. Decimal values are acceptable within the
+  // precision returned by Query/Precision.
+  string tradable_amount = 2;
+}
+
+// BasketCriteria defines a criteria by which credits can be added to a basket.
+message BasketCriteria {
+  
+  // filter defines condition(s) that credits should satisfy in order to be
+  // added to the basket.
+  Filter filter = 1;
+
+  // multiplier is an integer number which is applied to credit units when
+  // converting to basket units. For example if the multiplier is 2000, then
+  // 1.1 credits will result in 2200 basket tokens. If there are any fractional
+  // amounts left over in this calculation when adding credits to a basket,
+  // those fractional amounts will not get added to the basket.
+  string multiplier = 2;
```

Comment:
It looks like there are some conflicts that will occur here with #649. We are removing `BasketCriteria` and using `Filter` directly within `MsgCreateBasket` but I think this move was the right idea.

Reply:
good to know, let's see which one gets merged first!